src/input/composite_device/mod.rs

```diff
@@ -1404,38 +1404,45 @@ impl CompositeDevice {
             }
         }
 
-        let subsystem = device.subsystem();
-
-        let source_device = match subsystem.as_str() {
+        let dev_name = (&device).name();
+        let dev_subsystem = (&device).subsystem();
```

Comment:
this expression borrows a value the compiler would automatically borrow
```suggestion
        let dev_name = device.name();
        let dev_subsystem = device.subsystem();
```